core/predicate_check.go

```diff
@@ -15,6 +15,13 @@ import (
 
 // CheckPredicates checks that all precompile predicates are satisfied within the current [predicateContext] for [tx]
 func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.ProposerPredicateContext, tx *types.Transaction) error {
+	intrinsicGas, err := IntrinsicGas(tx.Data(), tx.AccessList(), tx.To() == nil, rules)
```

Comment:
Why do we check `IntrinsicGas` here in addition to tx pool and state transition? Do we want to return an early err?

Reply:
Yes, I'll add a comment to that effect.

Verifying the predicate costs some amount of computation, so we want to make sure that the transaction is capable of paying for that computation before performing it.